lib/topic_query.rb

```diff
@@ -897,20 +893,16 @@ def remove_muted_categories(list, user, opts = nil)
         list = list.where("topics.category_id IN (?)", category_ids) if category_ids.present?
       end
     elsif user
-      list = list.references("cu")
-        .where("
-        NOT EXISTS (
-          SELECT 1
-            FROM category_users cu
-           WHERE cu.user_id = :user_id
-             AND cu.category_id = topics.category_id
-             AND cu.notification_level = :muted
-             AND cu.category_id <> :category_id
-             AND (tu.notification_level IS NULL OR tu.notification_level < :tracking)
-        )", user_id: user.id,
-            muted: CategoryUser.notification_levels[:muted],
-            tracking: TopicUser.notification_levels[:tracking],
-            category_id: category_id || -1)
+      list = list
+        .references("cu")
+        .joins("LEFT JOIN category_users ON category_users.category_id = topics.category_id")
+        .where("category_users.user_id IS NULL OR category_users.user_id = :user_id", user_id: user.id)
+        .where("category_users.notification_level IS NULL
+               OR category_users.notification_level <> :muted
```

Comment:
Also here `COALESCE(category_users.notification_level, :regular) = :muted`. Additionally you should new `regular` param for this.

Reply:
thank you for your review,
I don't follow last comment
Original code was like that 
```
NOT EXISTS (
          SELECT 1
            FROM category_users cu
           WHERE cu.user_id = :user_id
             AND cu.category_id = topics.category_id
             AND cu.notification_level = :muted
             AND cu.category_id <> :category_id
             AND (tu.notification_level IS NULL OR tu.notification_level < :tracking)
```
so I want notification level different than muted
`category_users.notification_level <> :muted`

Reply:
Sorry! I have a typo in my previous comment. I like to change it like `COALESCE(category_users.notification_level, :regular) <> :muted` where the `:regular` is a new param `CategoryUser.notification_levels[:regular]`.

Reply:
OR in conditionals worry me a bit cause this is one area where SQL can get slow, but NOT EXISTS can be just as bad or worse.... did you time the full query this on some users with lots of state ?